### PR TITLE
Add an option for projectiles dealing AoE damage

### PR DESCRIPTION
**Pull request by ThisIsPIRI**
# Description
While making new weapons, I noticed there wasn't an option to create one that deals damage to all ships nearby when it hits one. So I added aoeRadius field to ProjectileConfig and changed the update method in Projectile to deal damage to all SolObjects within the radius if AoE is enabled(aoeRadius >= 0). I also added a couple of methods to SolMath and ObjectManager for this.

# Outstanding Work
- Maybe the Projectile itself, or the ships in the firing faction, should be excluded from the damage.
- Maybe the distance shouldn't be squared.

### Pre Pull Request Checklist:
- [x] Code has been scanned with [SonarLint](http://www.sonarlint.org/intellij/)
- [ ] Methods have been annotated with @ Nullable and @ Nonnull - isn't this outdated? The link tells me not to use these
- [x] There are no errors present in the project
- [x] Code has been formatted and indented
- [x] Methods have appropriate Javadoc ([How to write Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html))

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by vampcat** [comment]
@GooeyHub : ok to test

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/game/ObjectManager.java [hunk 1]
```diff
@@ -281,6 +282,20 @@ private void drawDebug0(GameDrawer drawer, SolGame game) {
         return myObjs;
     }
 
+    /**
+     * Returns all {@link SolObject}s whose squared distance to {@code fromObject} is less than {@code squaredDistance}.
+     * @return A {@code LinkedList<SolObject>} containing the matching objects.
+     */
+    public List<SolObject> getAllCloserThan(float squaredDistance, SolObject fromObject) { //Should the non-squared distance be used instead?
+        List<SolObject> result = new LinkedList<>();
```
Comment (Adrijaned) [inline engine/src/main/java/org/destinationsol/game/ObjectManager.java]: Why not use ArrayList here?

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/common/SolMath.java [hunk 2]
```diff
@@ -295,6 +295,17 @@ public static Vector2 distVec(Vector2 from, Vector2 to) {
...
 
+    /**
+     * Computes the squared scalar distance between two {@link Vector2 vectors}.
+     *
+     * @param from The 1st vector.
+     * @param to The 2nd vector.
+     * @return The distance between the two vectors in {@code float}.
+     */
+    public static float squaredScalarDist(Vector2 from, Vector2 to) {
+        return from.dst2(to);
```
[the remark this answers is not in the document]
Comment (ThisIsPIRI) [reply engine/src/main/java/org/destinationsol/common/SolMath.java]: Removed it.

**Comment by NicholasBatesNZ** [comment]
Thank you for this. It tests out fine!

**Pull request merged**